diff --git a/train.py b/train.py
index 401254f..b5cdea0 100644
--- a/train.py
+++ b/train.py
@@ -16,9 +16,12 @@ from torch.utils.data import DataLoader, RandomSampler, SequentialSampler
 from torch.utils.data.distributed import DistributedSampler
 from torch.utils.tensorboard import SummaryWriter
 from tqdm import tqdm
-
+import os
 from dataset.cifar import DATASET_GETTERS
 from utils import AverageMeter, accuracy
+import wandb
+wandb.init(entity='aml-audit', project="fix-match-pytorch", resume = True, sync_tensorboard = True)
+os.environ["WANDB_SILENT"] = "true"
 
 logger = logging.getLogger(__name__)
 best_acc = 0
@@ -313,7 +316,6 @@ def train(args, labeled_trainloader, unlabeled_trainloader, test_loader,
 
     labeled_iter = iter(labeled_trainloader)
     unlabeled_iter = iter(unlabeled_trainloader)
-
     model.train()
     for epoch in range(args.start_epoch, args.epochs):
         batch_time = AverageMeter()
@@ -438,7 +440,6 @@ def train(args, labeled_trainloader, unlabeled_trainloader, test_loader,
             logger.info('Best top-1 acc: {:.2f}'.format(best_acc))
             logger.info('Mean top-1 acc: {:.2f}\n'.format(
                 np.mean(test_accs[-20:])))
-
     if args.local_rank in [-1, 0]:
         args.writer.close()
 
